lib/widgets/achievementTracker_widget.dart

```diff
@@ -0,0 +1,105 @@
+import 'package:expansion_tile_card/expansion_tile_card.dart';
+import 'package:flutter/material.dart';
+import 'package:gra_terenowa/controller/tripData_controller.dart';
+import 'package:gra_terenowa/extras/colors.dart';
+import 'package:hive/hive.dart';
+
+class AchievementTracker extends StatefulWidget {
```

Comment:
Consider using StatelessWidget with GetX mechanism for tracking data. It will be consistent with the rest of the code.

Reply:
Could you provide a basic example on how to achieve this with GetX? From what I looked at, all data storage packages (including get_storage from GetX) return Future<> object when reading from cached files, which require using StatefulWidget to convert to a displayable, non-future object.

Reply:
It looks like GetStorage() doesn't need Future object and is very simple and lightweight. https://rrtutors.com/tutorials/flutter-getx-storage-shared-preferences

Reply:
To keep the momentum of our dev cycle I created a new issue for it and will close this PR.